### ISSUE TITLE
Tennis winner

### ISSUE THREAD

**Issue by penajay89** [VincentAuriau/Tennis-Prediction#85]

**Comment by penajay89** [comment]
import streamlit as st
import pandas as pd
from sklearn.linear_model import LogisticRegression

# Sample training data
data = [
    {"rank_diff": 3, "surface": "hard", "player1_win": 1},
    {"rank_diff": -2, "surface": "clay", "player1_win": 0},
    {"rank_diff": 0, "surface": "grass", "player1_win": 1},
    {"rank_diff": 1, "surface": "hard", "player1_win": 1},
    {"rank_diff": -5, "surface": "clay", "player1_win": 0},
    {"rank_diff": 4, "surface": "grass", "player1_win": 1},
    {"rank_diff": -1, "surface": "hard", "player1_win": 0},
    {"rank_diff": 2, "surface": "clay", "player1_win": 1},
    {"rank_diff": 0, "surface": "grass", "player1_win": 0},
    {"rank_diff": -3, "surface": "hard", "player1_win": 0},
]
df = pd.DataFrame(data)

# One-hot encode surface
df = pd.get_dummies(df, columns=["surface"], drop_first=True)

# Features and target
X = df.drop("player1_win", axis=1)
y = df["player1_win"]

# Train the model
model = LogisticRegression()
model.fit(X, y)

# Streamlit App
st.title("🎾 Tennis Winner Predictor")

# User input
player1_rank = st.number_input("Player 1 Rank", min_value=1, max_value=1000, value=5)
player2_rank = st.number_input("Player 2 Rank", min_value=1, max_value=1000, value=10)
surface = st.selectbox("Surface", options=["hard", "clay", "grass"])

# Create input for prediction
rank_diff = player2_rank - player1_rank
input_data = {
    "rank_diff": rank_diff,
    "surface_clay": 1 if surface == "clay" else 0,
    "surface_grass": 1 if surface == "grass" else 0,
}
input_df = pd.DataFrame([input_data])

# Prediction
prob = model.predict_proba(input_df)[0][1]
winner = "Player 1" if prob >= 0.5 else "Player 2"

st.subheader(f"Prediction: {winner} wins")
st.text(f"Win probability for Player 1: {prob:.2f}")
